PIL/GifImagePlugin.py

```diff
@@ -352,10 +352,21 @@ def _save(im, fp, filename, save_all=False):
 
         first_frame = None
         append_images = im.encoderinfo.get("append_images", [])
+        if "duration" in im.encoderinfo:
+            duration = im.encoderinfo["duration"]
+        else:
+            duration = None
+        frame_count = 0
         for imSequence in [im]+append_images:
             for im_frame in ImageSequence.Iterator(imSequence):
                 encoderinfo = im.encoderinfo.copy()
                 im_frame = _convert_mode(im_frame)
+                if duration is not None:
+                    if not isinstance(duration, list):
+                        encoderinfo["duration"] = duration
+                    else:
+                        encoderinfo["duration"] = duration[frame_count]
```

Comment:
Switch these around so the "if true" case comes first:
```
if isinstance(duration, list):
    encoderinfo["duration"] = duration[frame_count]
else:
    encoderinfo["duration"] = duration
```

Or, do we really need an explicit statement for when `duration` is an integer? Isn't that the old behaviour and dealt with already somewhere else?